### PR TITLE
Tutorial on custom mutations

### PR DESCRIPTION
**Pull request by MadeByMike**
That thing where you're like this will take an hour and 3 hours later you wrote a tutorial.

### CODE REVIEW

File: docs/guides/custom-mutations.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+# Custom Mutations
+
+Out of the box Keystone provides predictable CRUD operations (Create, Read, Update and Delete) for Lists. The generated graphQL queries and mutations are the primary method for updating data in a List. These should be enough for most application requirements.
+
+Custom Queries and Mutations may be required if you wish to preform non-CRUD operations or actions that don't relate to a specific List.
+
+See the [GraphQL Philosophy](https://v5.keystonejs.com/guides/graphql-philosophy) for more information on how Keystone implements CRUD operations in GraphQL and when Custom Queries and Mutations may be required.
+
+You can add to Keystone's generated schema with custom types, queries, and mutations using the `keystone.extendGraphQLSchema()` method.\
```
Comment (sarneaud) [inline docs/guides/custom-mutations.md]: Wayward `\`.

File: docs/guides/custom-mutations.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+# Creating an Custom Mutation
+
+A common example where a custom mutation might be beneficial is if you want to increment a value.
+
+Like any problem there are multiple solutions. You can implement an incrementing value with (Hooks)[https://v5.keystonejs.com/guides/hooks] but in this example we're going to look at how to do this with a custom mutation.
```
Comment (sarneaud) [inline docs/guides/custom-mutations.md]: `()` and `[]` swapped in link.

File: docs/guides/custom-mutations.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    ...oldItem,
+    views: (oldItem.views || 0) + 1,
+  });
+  return newItem;
+};
+```
+
+Note: The value of `views` may be `undefined` initially, so before we increment it we make sure to convert any `falsey` values to `0`.
+
+Our custom mutation is now available to the client that trigger it with the following graphQL query:
```
Comment (sarneaud) [inline docs/guides/custom-mutations.md]: Is this sentence missing a word or something?

File: docs/guides/custom-mutations.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+  mutations: [
+    {
+      schema: `incrementPageViews(id: ID!): Page`,
+      resolver: incrementPageViews,
+    },
+  ],
+});
+```
+
+The last step is to define the resolver function `incrementPageViews`.
```
Comment (sarneaud) [inline docs/guides/custom-mutations.md]: Is this sentence meant to go a couple of paragraphs down?

File: docs/guides/custom-mutations.md (hunk 1 above)
Comment (MadeByMike) [inline docs/guides/custom-mutations.md]:
```suggestion
You can add to Keystone's generated schema with custom types, queries, and mutations using the `keystone.extendGraphQLSchema()` method.
```

File: docs/guides/custom-mutations.md (hunk 2 above)
Comment (MadeByMike) [inline docs/guides/custom-mutations.md]:
```suggestion
Like any problem there are multiple solutions. You can implement an incrementing value with [Hooks](https://v5.keystonejs.com/guides/hooks) but in this example we're going to look at how to do this with a custom mutation.
```

File: docs/guides/custom-mutations.md (hunk 4 above)
Comment (MadeByMike) [inline docs/guides/custom-mutations.md]:
```suggestion
```

File: docs/guides/custom-mutations.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+      access: {
+        create: false,
+        read: true,
+        update: false,
+      },
+    },
+  },
+};
+```
+
```
Comment (MadeByMike) [inline docs/guides/custom-mutations.md]:
```suggestion

The last step is to define the resolver function `incrementPageViews`.

```

File: docs/guides/custom-mutations.md (hunk 3 above)
Comment (MadeByMike) [inline docs/guides/custom-mutations.md]:
```suggestion
Our custom mutation is now available to the client that can use it to increment page views like this:
```

File: docs/guides/custom-mutations.md [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+Note: The value of `views` may be `undefined` initially, so before we increment it we make sure to convert any `falsey` values to `0`.
+
+Our custom mutation is now available to the client that trigger it with the following graphQL query:
+
+```
+mutation incrementPageViews($id: ID!) {
+  mutation {
+    incrementPageViews(id: $id) {
+      views
+  }
```
[the remark this answers is not in the document]
Comment (MadeByMike) [reply docs/guides/custom-mutations.md]: views is an integer

**Pull request merged**